tendrl/node_agent/node_sync/__init__.py

```diff
@@ -164,10 +153,40 @@ def run(self):
                 sync_cluster_contexts_thread.daemon = True
                 sync_cluster_contexts_thread.start()
                 sync_cluster_contexts_thread.join()
-
+            # Update node alert count
+            if not NS.tendrl.objects.NodeAlertCounters().exists():
```

Comment:
You should be increment the alert count if NS.tendrl.objects.NodeAlertCounters().exists():  

right?

Reply:
@r0h4n When node details are deleted by ttl or initially if alert count object is not present so at that time only it will save alert count object in etcd.

Reply:
What i feel is let it calculate alert count in every sync because it calculating alert count only for that node only, it is not reading all alerts. So whatever happens alert count won't goes wrong

@shtripat @r0h4n

Reply:
just a suggestion